vexim/userchangepassword.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/*
+  Endpoint for external interface like roundcube to change the users password
+*/
+
+include_once dirname(__FILE__) . '/config/variables.php';
+include_once dirname(__FILE__) . '/config/functions.php';
+
+# first check if we have sufficient post variables to achieve a successful login... if not the login fails immediately
+if (!isset($_POST['user']) || $_POST['user']==='' 
+    || !isset($_POST['curpass']) || $_POST['curpass']===''
+    || !isset($_POST['newpass']) || $_POST['newpass']===''){
+    http_response_code(400);
+    die("invalid arguments");
+}
+
+# sql statement based on username
+$query = "SELECT u.user_id,u.username,u.crypt
+FROM users as u JOIN domains as d ON d.domain_id = u.domain_id
+WHERE u.username=:username
+AND u.enabled = 1
+AND d.enabled = 1
+AND u.type IN ('local', 'piped')";
+$sth = $dbh->prepare($query);
+$success = $sth->execute(array(':username'=>$_POST['user']));
+if(!$success) {
```

Comment:
I don't think we should return a success status code here.

Reply:
This is a code snipped from the original login, the sql query is a bit modded to nerrow by type.

Reply:
That doesn't mean it can't be improved, especially since this script will act as a RESTful API endpoint of sorts.